app/services/get_pending_data_sharing_agreements_for_provider_user.rb

```diff
@@ -0,0 +1,7 @@
+class GetPendingDataSharingAgreementsForProviderUser
+  def self.call(provider_user:)
+    provider_user.providers.left_outer_joins(:provider_agreements).where(provider_agreements: {id: nil}).map do |provider|
```

Comment:
Nice 👍

Reply:
It's even better now (one sql query for everything)